apps/web/src/components/Publication/OpenAction/UnknownModule/Tip/index.tsx

```diff
@@ -28,6 +32,25 @@ const TipOpenAction: FC<TipOpenActionProps> = ({
   const targetPublication = isMirrorPublication(publication)
     ? publication?.mirrorOn
     : publication;
+  const [hasTipped, setHasTipped] = useState(false);
+  const [countTips, setCountTips] = useState<number>(0);
+
+  const { data: tipData } = useUnknownOpenActionDataQuery({
+    variables: {
+      module: VerifiedOpenActionModules.Tip,
+      pubId: targetPublication.id
+    }
+  });
+
+  useEffect(() => {
+    setHasTipped(
+      tipData?.publication?.__typename === 'Post' &&
+        Boolean(tipData.publication.operations.actedOn.length)
+    );
+    if (tipData?.publication?.__typename === 'Post') {
+      setCountTips(tipData.publication.stats.countOpenActions);
+    }
+  }, [tipData]);
```

Comment:
We can move this on `onCompleted` callback under `useUnknownOpenActionDataQuery`

Reply:
Also do we need state for this? can we directly render from the `tipData`? no?

Reply:
> Also do we need state for this? can we directly render from the `tipData`? no?

Good point. I'm still a little new to React and hooks and forgot that there's no race condition since `tipData` will actually trigger the query. 🙂 Changes submitted in 29bd91d.